csi/moac/index.js

```diff
@@ -205,6 +207,14 @@ async function main () {
     csiServer.makeReady(registry, volumes);
     log.info('MOAC is warmed up and ready to 🚀');
   }, warmupSecs * 1000);
+
+  if (!opts.s) {
+    pvOper = new PvOperator(
+      kubeConfig,
+      volumes
+      )
+    pvOper.start()
+  }
```

Comment:
you should move the PV operator initialization to line 201 in the new file. Otherwise you start the operator before warmup which means that the picture of the cluster that moac has may not be complete and weird things can happen then.